src/uuid.cr

```diff
@@ -0,0 +1,81 @@
+require "secure_random"
+require "./uuid/*"
+
+# Universally Unique IDentifier.
+#
+# Supports custom variants with arbitrary 16 bytes as well as (RFC 4122)[https://www.ietf.org/rfc/rfc4122.txt] variant
+# versions.
+struct UUID
+  # Internal representation.
+  @bytes : StaticArray(UInt8, 16)
+
+  # Generates RFC 4122 v4 UUID.
+  def initialize
+    bytes = StaticArray(UInt8, 16).new(0_u8)
+    bytes.to_unsafe.copy_from SecureRandom.random_bytes(16).pointer(16), 16
+    initialize bytes
```

Comment:
We shouldn't call `initialize` itself. In Crystal we use a `new` factory instead. For example:

```crystal
def self.new
  bytes = ...
  new bytes
end
```